src/Execution/ResolveInfo.php

```diff
@@ -42,7 +42,7 @@ public function __construct(
      * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder<TModel>|\Illuminate\Database\Eloquent\Relations\Relation<TModel>|\Laravel\Scout\Builder  $builder
      * @param  array<string>  $scopes
      * @param  array<string, mixed>  $args
-     * @param  (callable(\Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective): bool)|null  $directiveFilter
+     * @param  (callable(\Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective|\Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null  $directiveFilter
```

Comment:
```suggestion
     * @param  (callable(\Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective): bool)|(callable(\Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null  $directiveFilter
```

Reply:
Phpstan will report

```
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Execution/ResolveInfo.php
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  55     Parameter #1 $directiveFilter of method Nuwave\Lighthouse\Scout\ScoutEnhancer<TModel of Illuminate\Database\Eloquent\Model>::enhanceBuilder() expects (callable(Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null, (callable(Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective):
         bool)|(callable(Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null given.
         ✏️  src/Execution/ResolveInfo.php:55
  58     Parameter #3 $directiveFilter of static method Nuwave\Lighthouse\Execution\ResolveInfo::applyArgBuilderDirectives() expects (callable(Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective): bool)|null, (callable(Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective):
         bool)|(callable(Nuwave\Lighthouse\Scout\ScoutBuilderDirective): bool)|null given.
         ✏️  src/Execution/ResolveInfo.php:58
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

Reply:
> I think we can just ignore the specific errors you posted, and use the more accurate type.

Real code will fail, you can try following test. 

```php
public function testEnhanceBuilder(): void {
        $directiveOne = new class() implements Directive, ArgBuilderDirective {
            public static function definition(): string {
                return '';
            }

            public function handleBuilder(
                QueryBuilder|EloquentBuilder|Relation $builder,
                mixed $value,
            ): QueryBuilder|EloquentBuilder|Relation {
                return $builder->where('one', $value);
            }
        };
        $directiveTwo = new class() implements Directive, ScoutBuilderDirective {
            public static function definition(): string {
                return '';
            }

            public function handleScoutBuilder(ScoutBuilder $builder, mixed $value): ScoutBuilder {
                return $builder->where('two', $value);
            }
        };

        $value            = 'value';
        $argumentSet      = new ArgumentSet();
        $argumentSet->directives = new Collection();

        $argumentSearch = new Argument();
        $argumentSearch->value = '*';
        $argumentSearch->directives->push(
            new SearchDirective(),
        );
        $argumentSet->arguments['search'] = $argumentSearch;

        $argumentA        = new Argument();
        $argumentA->value = $value;
        $argumentA->directives->push(
            $directiveTwo,
            $directiveOne,
        );
        $argumentSet->arguments['arg'] = $argumentA;

        $builder = new ScoutBuilder(new User(), '*');
        $resolveInfo              = new class() extends ResolveInfo {
            /** @phpstan-ignore-next-line no need to call parent `__construct` */
            public function __construct() {
                // empty
            }
        };
        $resolveInfo->argumentSet = $argumentSet;

        $resolveInfo->enhanceBuilder(
            $builder,
            [],
            null,
            [],
            $this->createMock(GraphQLContext::class),
            $resolveInfo,
            static fn (ArgBuilderDirective $directive): bool => true,
        );
    }
```

```
TypeError: Tests\Unit\Execution\ResolveInfoTest::Tests\Unit\Execution\{closure}(): Argument #1 ($directive) must be of type Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective, Nuwave\Lighthouse\Support\Contracts\Directive@anonymous given, called in /project/tmp/lighthouse/src/Scout/ScoutEnhancer.php on
 line 80
```

So probably phpstan is right in this case and `(callable(ArgBuilderDirective|ScoutBuilderDirective): bool)|null` is the right type.

Reply:
> `(callable(ArgBuilderDirective|ScoutBuilderDirective): bool)|null` is the right type.

With it, phpstan will report that:

```
 67     Parameter #7 $directiveFilter of method Nuwave\Lighthouse\Execution\ResolveInfo::enhanceBuilder() expects (callable(Nuwave\Lighthouse\Scout\ScoutBuilderDirective|Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective): bool)|null, Closure(Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective):  
         true given.
         💡 Type Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective of parameter #1 $directive of passed callable needs to be same or wider than parameter type Nuwave\Lighthouse\Scout\ScoutBuilderDirective|Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective of accepting callable.
         ✏️  tests/Unit/Execution/ResolveInfoTest.php:67
```